File: rank_test_problems_b_0002/rank_test_problems_b_0002__3.cs
// 実行時間 20ms
using static System.Console;
using static System.Linq.Enumerable;

internal class Program
{
    static void Main()
    {
        int[] nq = ReadLine().Split().Select( int.Parse ).ToArray();
        int[] b = new int[nq[0]];

        foreach( int _ in Range( 0, nq[1] ) ) {
            string[] s = ReadLine().Split();
            int i = int.Parse( s[1] ) - 1;

            if( b[i] == 0 && s[0][0] == 'A' ) {
                b[i] = 1;
            } else if( b[i] == 0 && s[0][0] == 'B' ) {
                b[i] = -1;
            }
        }
        WriteLine( 0 < b.Sum() ? "A" : b.Sum() < 0 ? "B" : "Draw" );
        return;
    }
}

File: rank_test_problems_b_0002/rank_test_problems_b_0002__3.py
# 実行時間 10ms
n, q = list(map(int, input().split()))
b = [0] * n

for _ in [0] * q:
    p, x = input().split()
    x = int(x) - 1

    if b[x] == 0 and p == "A":
        b[x] = 1
    elif b[x] == 0 and p == "B":
        b[x] = -1
print("A" if 0 < sum(b) else "B" if sum(b) < 0 else "Draw")
